packages/slice-machine/components/DeleteCTModal/index.tsx

```diff
@@ -104,11 +110,14 @@ export const DeleteCustomTypeModal: React.FunctionComponent<
             <Button
               label="Delete Locally"
               variant="danger"
-              // eslint-disable-next-line @typescript-eslint/no-misused-promises
-              onClick={async () => {
-                await deleteCustomType(customType?.local.id as string);
-                closeDeleteCustomTypeModal();
-              }}
+              isLoading={isDeletingCustomType}
+              onClick={() =>
+                deleteCustomType(
+                  customType?.local.id ?? "",
```

Comment:
It should never happen, but do you know what happens if `customType` is actually undefined here?